### ISSUE TITLE
Patch is generated for wrong pod version

### ISSUE THREAD

**Issue by sryze** [DoubleSymmetry/cocoapods-patch#12]
It seems that cocoapods-patch creates the diff for the latest stable version of the specified pod. 

For example, in my iOS project I have the [JWT](https://github.com/yourkarma/JWT) pod installed version 3.0.0-beta.13, and I have modified it by adding a couple of lines in `JWTCryptSecurity+ErrorHandling.m` like so:

![Screen Shot 2021-07-06 at 15 02 42](https://user-images.githubusercontent.com/4525736/124573324-4119aa00-de6b-11eb-819e-ba4b84c501a8.png)

However, the generated patch contains more changes, as if it was created against version 2.1.0 of JWT (see attached JWT.diff.zip).

[JWT.diff.zip](https://github.com/DoubleSymmetry/cocoapods-patch/files/6768789/JWT.diff.zip)

I used the following command to create the patch:

```
pod patch create JWT
```

CocoaPods version: 1.10.1

Inside `Podfile.lock` it says:

```
- JWT (3.0.0-beta.13):
```

My Podfile however does not specify the version of JWT as it's a dependency of another pod (CodePush).

**Comment by dcvz** [comment]
interesting @sryze -- it could be possible we're using the wrong version.. I'll try to take a look -- in the meantime are you able to modify the patch file down to just the change you actually want?

**Comment by dcvz** [comment]
I took a deeper look at this @sryze -- its definitely using the right version of the pod as in `2.1.0` the files you edited didn't exist yet. I'm wondering why it's pulling in more changes though, it could be Xcode modifying something or perhaps the pod install process.

I've added some more safety around this though, to make sure to respect your lock file when comparing to make the diff in #15 . It'll be out soon with the next release.
